File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize){
    int profit = 0;
    int start_price = prices[0];
    
    for(int i = 0; i < pricesSize; i++){
        profit = fmax(profit, prices[i]-start_price);
        start_price = fmin(start_price, prices[i]);
    }
    return profit;
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        profit = 0
        start_price = prices[0]
        for i in range(1, len(prices)):
            profit = max(profit, (prices[i]-start_price))
            start_price = min(start_price, prices[i])
        return profit